compiler/surface/lexer_fr.cppo.ml

```diff
@@ -0,0 +1,141 @@
+#define MS_BOOLEAN "bool�en"
+#define MS_SUM "somme"
+#define MS_FILLED "rempli"
+#define MS_DEFINITION "d�finition"
+#define MS_LABEL "�tiquette"
+#define MS_EXCEPTION "exception"
+#define MS_DEFINED_AS "�gal �"
+#define MR_DEFINED_AS "�gal", space_plus, "�"
+#define MS_MATCH "selon"
+#define MS_WILDCARD "n'importe quel"
+#define MR_WILDCARD "n'importe", space_plus, "quel"
+#define MS_WITH "sous forme"
+#define MR_WITH "sous", space_plus, "forme"
+#define MS_UNDER_CONDITION "sous condition"
+#define MR_UNDER_CONDITION "sous", space_plus, "condition"
+#define MS_IF "si"
+#define MS_THEN "alors"
+#define MS_ELSE "sinon"
+#define MS_CONDITION "condition"
+#define MS_CONTENT "contenu"
+#define MS_STRUCT "structure"
+#define MS_ASSERTION "assertion"
+#define MS_VARIES "varie"
+#define MS_WITH_V "avec"
+#define MS_FOR "pour"
+#define MS_ALL "tout"
+#define MS_WE_HAVE "on a"
+#define MR_WE_HAVE "on", space_plus, "a"
+#define MS_FIXED "fix�"
+#define MS_BY "par"
+#define MS_RULE "r�gle"
+#define MS_EXISTS "existe"
+#define MS_IN "dans"
+#define MS_SUCH "tel"
+#define MS_THAT "que"
+#define MS_AND "et"
+#define MS_OR "ou"
+#define MS_XOR "ou bien"
+#define MR_XOR "ou", space_plus, "bien"
+#define MS_NOT "non"
+#define MS_MAXIMUM "maximum"
+#define MS_MINIMUM "minimum"
+#define MS_FILTER "filtre"
+#define MS_MAP "application"
+#define MS_INIT "initial"
+#define MS_CARDINAL "nombre"
+#define MS_YEAR "an"
+#define MS_MONTH "mois"
+#define MS_DAY "jour"
+#define MS_TRUE "vrai"
+#define MS_FALSE "faux"
+
+#define MR_MONEY_OP_SUFFIX 0x20AC (* The euro sign *)
+
+(* Builtins *)
+
+#define MS_IntToDec "entier_vers_d�cimal"
+#define MS_GetDay "acc�s_jour"
+#define MS_GetMonth "acc�s_mois"
+#define MS_GetYear "acc�s_ann�e"
+
+(* Directives *)
+
+#define MR_BEGIN_METADATA "D�but", Plus hspace, "m�tadonn�es"
+#define MR_END_METADATA "Fin", Plus hspace, "m�tadonn�es"
+#define MR_LAW_INCLUDE "Inclusion"
+#define MX_AT_PAGE \
+   '@', Star hspace, "p.", Star hspace, Plus digit -> \
+      let s = Utf8.lexeme lexbuf in \
+      let i = String.index s '.' in \
+      AT_PAGE (int_of_string (String.trim (String.sub s i (String.length s - i))))
+
+(* More complex cases *)
+
+#define MX_MONEY_AMOUNT \
+   digit, Star (digit | hspace), Opt (',', Rep (digit, 0 .. 2)), Star hspace, 0x20AC -> \
+      let extract_parts = R.regexp "([0-9]([0-9 ]*[0-9]|))(,([0-9]{0,2})|)" in \
+      let str = Utf8.lexeme lexbuf in \
+      let parts = R.get_substring (R.exec ~rex:extract_parts str) in \
+      (* Integer literal*) \
+      let units = parts 1 in \
+      let remove_spaces = R.regexp " " in \
+      let units = \
+        Runtime.integer_of_string (R.substitute ~rex:remove_spaces ~subst:(fun _ -> "") units) \
+      in \
+      let cents = \
+        try Runtime.integer_of_string (parts 4) with Not_found -> Runtime.integer_of_int 0 \
+      in \
+      L.update_acc lexbuf; \
+      MONEY_AMOUNT (units, cents)
+
+#define MX_DECIMAL_LITERAL \
+   Plus digit, ',', Star digit -> \
```

Comment:
Hm actually we could define a macro just for the decimal separator... couldn't we ?

Reply:
No, in English the decimal separator is `.` instead of `,` in France.

Reply:
yes, what I mean is that we could factorise the digit parsing and have a macro only for `.` or `,`